GVRf/Framework/jni/gl/gl_texture.h

```diff
@@ -56,8 +56,10 @@ class GLTexture : public GLPendingTask {
         memcpy(texture_parameters_, texture_parameters, sizeof(int) * 5);
     }
 
-    ~GLTexture() {
-        deleter_->queueTexture(id_);
+    virtual ~GLTexture() {
+        if (GL_TASK_NONE == pending_gl_task_) {
```

Comment:
id_ is a GLuint. glGenTextures can give you any GLuint. There is no invalid value to set it to strictly speaking. Instead of a dedicated flag I chose what I thought was the lesser evil.